models/authentication/agent/agent-connected/index.tsx

```diff
@@ -92,7 +93,28 @@ export class AgentConnected {
   }
 
   async getAgentHabilitation() {
-    const superAgentScopes = await superAgentsList.getScopeForAgent(this.email);
+    // Get all groups this agent belongs to from D-Roles API
+    const drolesGroups = await getGroupsByEmail(this.email);
+
+    // Each group has a space-separated string of scopes
+    // Extract all scopes from all groups and flatten into single array
+    const superAgentScopesWithDuplicates = drolesGroups
+      .map((group) => {
+        return group.scopes.split(' ') as IAgentScope[];
+      })
+      .flat();
+
+    // Get additional scopes from S3 storage for this agent
+    const superAgentScopesFromS3 = await superAgentsList.getScopeForAgent(
+      this.email
+    );
+
+    // Combine scopes from both D-Roles groups and S3 storage
+    // Remove any duplicate scopes using Set
+    const superAgentScopes = [
+      ...new Set(superAgentScopesWithDuplicates),
+      ...new Set(superAgentScopesFromS3),
+    ];
```

Comment:
Merging a list of scopes is also done line 102. Maybe create a helper function in `../scopes`that is used to merge scope and unit tested as this is core business logic